From d9ecdc8e070b34cad60550eacc1eda4e8b8287e3 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 26 Jun 2023 09:10:04 +0200
Subject: [PATCH 4/4] replace wsprintf with snprintf

wsprintf is deprecated and not allowed in Universal Windows Platform builds.
It can be replaced by StringCbPrintfA but it requires strsafe.h. We can
just use snprintf() which is the C runtime equivalent.
---
 src/load_mod.cpp | 6 +++---
 src/sndfile.cpp  | 3 +--
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/load_mod.cpp b/src/load_mod.cpp
index 9ea48c8..b4c11b7 100644
--- a/src/load_mod.cpp
+++ b/src/load_mod.cpp
@@ -92,7 +92,7 @@ WORD CSoundFile::ModSaveCommand(const MODCOMMAND *m, BOOL bXM) const
 	case CMD_TONEPORTAVOL:		command = 0x05; break;
 	case CMD_VIBRATOVOL:		command = 0x06; break;
 	case CMD_TREMOLO:			command = 0x07; break;
-	case CMD_PANNING8:			
+	case CMD_PANNING8:
 		command = 0x08;
 		if (bXM)
 		{
@@ -328,7 +328,7 @@ BOOL CSoundFile::ReadMod(const BYTE *lpStream, DWORD dwMemLength)
 	}
 	if ((dwWowTest < 0x600) || (dwWowTest > dwMemLength)) nErr += 8;
 	if ((m_nSamples == 15) && (nErr >= 16)) return FALSE;
-	// Default settings	
+	// Default settings
 	m_nType = MOD_TYPE_MOD;
 	m_nDefaultSpeed = 6;
 	m_nDefaultTempo = 125;
@@ -471,7 +471,7 @@ BOOL CSoundFile::SaveMod(LPCSTR lpszFileName, UINT nPacking)
 	if (m_nChannels == 4)
 		strcpy((LPSTR)&bTab, "M.K.");
 	else
-		wsprintf((LPSTR)&bTab, "%luCHN", m_nChannels);
+		snprintf(LPSTR)&bTab, sizeof(bTab), "%luCHN", m_nChannels);
 	fwrite(bTab, 4, 1, f);
 	// Writing patterns
 	for (UINT ipat=0; ipat<nbp; ipat++) if (Patterns[ipat])
diff --git a/src/sndfile.cpp b/src/sndfile.cpp
index f5189ff..e34f81e 100644
--- a/src/sndfile.cpp
+++ b/src/sndfile.cpp
@@ -369,7 +369,6 @@ void CSoundFile::FreeSample(LPVOID p)
 	}
 }
 
-
 //////////////////////////////////////////////////////////////////////////
 // Misc functions
 
@@ -383,7 +382,7 @@ void CSoundFile::ResetMidiCfg()
 	strcpy(&m_MidiCfg.szMidiGlb[MIDIOUT_NOTEOFF*32], "9c n 0");
 	strcpy(&m_MidiCfg.szMidiGlb[MIDIOUT_PROGRAM*32], "Cc p");
 	strcpy(&m_MidiCfg.szMidiSFXExt[0], "F0F000z");
-	for (int iz=0; iz<16; iz++) wsprintf(&m_MidiCfg.szMidiZXXExt[iz*32], "F0F001%02X", iz*8);
+	for (int iz=0; iz<16; iz++) snprintf(&m_MidiCfg.szMidiZXXExt[iz*32], sizeof(m_MidiCfg.szMidiZXXExt), "F0F001%02X", iz*8);
 }
 
 
-- 
2.37.3.windows.1

